### ISSUE TITLE
icloud integration broken now (January 2023)?

### ISSUE THREAD

**Issue by jsolderitsch** [kolbyjack/MMM-Wallpaper#83]
I can use MMM-Wallpaper for bing and apod ok.

I have an icloud public album with this ID: ```0ae2R2FLz_K0nxDsusXUnhbKQ```

I enter this config section in my config.js:
```
                {
                          module: "MMM-Wallpaper",
                          position: "fullscreen_below",
                          config: { // See "Configuration options" for more information.
                                source: "icloud:0ae2R2FLz_K0nxDsusXUnhbKQ",
                             // source: "apod",
                             // nasaApiKey: "YBwBlmYDD6CZslRexFvHIhIwveoeM0IicK6rdYE7",
                             slideInterval: 60 * 1000 // Change slides every minute
                           }
                },
```
I get this error in the magic mirror log:
```
0|mm       | [4254:0105/112838.479320:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
0|mm       | [4156:0105/112838.550715:ERROR:command_buffer_proxy_impl.cc(128)] ContextResult::kTransientFailure: Failed to send GpuControl.CreateCommandBuffer.
0|mm       | [05.01.2023 11:28.41.884] [ERROR] (node:4108) UnhandledPromiseRejectionWarning: FetchError: request to https://p2272-sharedstreams.icloud.com/0ae2R2FLz_K0nxDsusXUnhbKQ/sharedstreams/webstream failed, reason: getaddrinfo ENOTFOUND p2272-sharedstreams.icloud.com
0|mm       |     at ClientRequest.<anonymous> (/home/mirror/MagicMirror/node_modules/node-fetch/lib/index.js:1491:11)
0|mm       |     at ClientRequest.emit (node:events:513:28)
0|mm       |     at TLSSocket.socketErrorListener (node:_http_client:481:9)
0|mm       |     at TLSSocket.emit (node:events:513:28)
0|mm       |     at emitErrorNT (node:internal/streams/destroy:157:8)
0|mm       |     at emitErrorCloseNT (node:internal/streams/destroy:122:3)
0|mm       |     at processTicksAndRejections (node:internal/process/task_queues:83:21)
```
Indeed the url with p2272 beginning it is "not found" by my browser.

Any fixes to suggest?

**Comment by kolbyjack** [comment]
If you look at your album properties, is `https://www.icloud.com/sharedalbum/#0ae2R2FLz_K0nxDsusXUnhbKQ` what it shows for the "Anyone can view this album at:" link?  That doesn't look like other public album ids that I've seen.

**Comment by jsolderitsch** [comment]
That URL takes me to the apple.com/icloud page. Without any login.

I have an iCloud Links item as seen:
![Screen Shot 2023-01-05 at 1 36 17 PM](https://user-images.githubusercontent.com/23091236/210855350-5271e781-af86-43c2-9461-8d97b46d478a.png)

I do not see a "sharedalbum" item at all in the main iCloud interface.

I created the shared album manually from the iCloud Photos page and this generated the link URL:

[](https://share.icloud.com/photos/0ae2R2FLz_K0nxDsusXUnhbKQ)

If I open this, it resolves to:
[](https://www.icloud.com/photos/#0ae2R2FLz_K0nxDsusXUnhbKQ)

So I don't have a property inspector for the created album. At least I don't see one.

**Comment by kolbyjack** [comment]
I don't see a way to manage the sharing settings from the website, if you use the Photos app on your iPhone, and open the album there, you should see the link.

**Comment by jsolderitsch** [comment]
Ah, not on the iCloud website, but on the Photos app on my Mac, there is a property inspector and I was able to allow for public sharing. Let me try to integrate this. The album ID is different as is the sharable URL:

[https://www.icloud.com/sharedalbum/#B0g532ODW3ZgPo](url)

I will experiment and report back.

**Comment by jsolderitsch** [comment]
Yes, I was not aware of the Photos app integration this way. I was using the Mac and not my iPhone. But using the photos app on the Mac  does work as well. I now have it working using the album ID from the app. I use the iPhone to take photos but have not tried to share them in this way before. So MMM-Wallpaper is not broken if you use the Shared Album feature from the app.
